core/src/main/scala/chisel3/Module.scala

```diff
@@ -392,6 +392,20 @@ package experimental {
     /** Internal check if a Module's constructor has finished executing */
     private[chisel3] def isClosed = _closed
 
+    /** Mutable state that indicates if IO is allowed to be created for this module.
+      * This can be used for advanced Chisel library APIs that want to limit
+      * what IO is allowed to be created for a module.
+      */
+    private var _allowIOCreation = true
+
+    /** If true, then this module is allowed to have user-created IO. */
+    private[chisel3] def allowIOCreation = _allowIOCreation
```

Comment:
Rename to "isIOCreationAllowed" ? The "allowIOCreation" vs "disallowIOCreation()" is confusing because one is a setter, while the other is a getter, but their names imply inverses of each other (e.g. isEmpty vs nonEmpty)

Reply:
Makes sense. Good suggestion. I updated to this in the rebase.